### ISSUE TITLE
c++ Windows  compile error 

### ISSUE THREAD

**Issue by sikuntiannan** [pytorch/pytorch#18645]
visual studio2017  ；Reference directory  O:\pytorch\libtorch\include\torch\csrc\api\include  
  O:\pytorch\libtorch\include   library directory O:\pytorch\libtorch\lib .... 
At the 101 of document rnn.h. There's a mistake.have error :cast to function type is illegal
 pytorch is https://download.pytorch.org/libtorch/cu100/libtorch-win-shared-with-deps-latest.zip
cuda is 10.0
windwos 10
visual studio 2017
release x64
cl.exe 14.16.27023

I think this is a compiler error. To this, there are Windows installation tutorials, you can compare the difference. Select the compilation environment and version.

**Comment by xsacha** [comment]
Do you also get the error in the nightly build?
Looks like this was fixed already.

**Comment by sikuntiannan** [comment]
no i in the stable 1.0. isn`t nightly .
what it is" Looks like this was fixed already".

**Comment by xsacha** [comment]
I mean, I think this error did exist but it no longer exists in the latest master.
To confirm, can you try the nightly build?

**Comment by sikuntiannan** [comment]
i try.

**Comment by asa008** [comment]
is this problem solved now? I compiled the source code yesterday, but still encounter the problem.

**Comment by gchanan** [comment]
@asa008 can you include a log of your build?

**Comment by asa008** [comment]
error occured in rnn.h:
  // The function signature of `rnn_relu`, `rnn_tanh` and `gru`. 
  using RNNFunctionSignature = std::tuple<Tensor, Tensor>(     
     /*input=*/const Tensor&,
      /*state=*/const Tensor&,
      /*params=*/TensorList,
      /*has_biases=*/bool,
      /*layers=*/int64_t,
      /*dropout=*/double,
      /*train=*/bool,
      /*bidirectional=*/bool,
      /*batch_first=*/bool);  
	  
then, according to https://discuss.pytorch.org/t/c2066-error-in-rnn-h/37150/2  
to modify above as following, solved:

using RNNFunctionSignature = std::function<std::tuple<Tensor, Tensor>(
      /*input=*/const Tensor&,
      /*state=*/const Tensor&,
      /*params=*/TensorList,
      /*has_biases=*/bool,
      /*layers=*/int64_t,
      /*dropout=*/double,
      /*train=*/bool,
      /*bidirectional=*/bool,
      /*batch_first=*/bool)>;

**Comment by ezyang** [comment]
@asa008 Can you paste what the contents of `generic_forward` (right below) are? It should look like this:

```
  /// A generic `forward()` used for RNN and GRU (but not LSTM!). Takes the ATen
  /// RNN function as first argument.
  RNNOutput generic_forward(
      std::function<RNNFunctionSignature> function,
      const Tensor& input,
      Tensor state);
```

Can you paste the exact contents of your error message?

**Comment by vladimir-aubrecht** [comment]
Closing issue as it's stale for 4 years and cannot reproduce issue with latest stack. Feel free to open new issue if problem persist with latest version.
